docs/apache-airflow/core-concepts/executor/index.rst

```diff
@@ -41,32 +49,175 @@ If you want to check which executor is currently set, you can use the ``airflow
+* :doc:`BatchExecutor <apache-airflow-providers-amazon:executors/batch-executor>`
+
+
 
+*Containerized Executors*
+
+Airflow tasks are executed ad hoc inside containers/pods. Each task is isolated in its own containerized environment that is deployed when the Airflow task is queued.
+
+**Pros**: Each Airflow task is isolated to one container so no noisy neighbor problem. The execution environment can be customized for specific tasks (system libs, binaries, dependencies, amount of resources, etc). Cost effective as the workers are only alive for the duration of the task.
+
+**Cons**: There is latency on startup since the container or pod needs to deploy before the task can begin. Can be expensive if you're running many short/small tasks. No workers to manage however you must manage something like a Kubernetes cluster.
+
+**Examples**:
+
+* :doc:`KubernetesExecutor <apache-airflow-providers-cncf-kubernetes:kubernetes_executor>`
+* :doc:`EcsExecutor <apache-airflow-providers-amazon:executors/ecs-executor>`
 
 .. note::
 
     New Airflow users may assume they need to run a separate executor process using one of the Local or Remote Executors. This is not correct. The executor logic runs *inside* the scheduler process, and will run the tasks locally or not depending the executor selected.
 
+Using Multiple Executors Concurrently
+-------------------------------------
+
+.. warning::
+    Multiple executor configuration is an alpha/experimental feature at the moment and may be subject to change without warning.
+
+Starting with version 2.10.0, Airflow can now operate with a multi-executor configuration. Each executor has its own set of pros and cons, often they are trade-offs between latency, isolation and compute efficiency among other properties (see :ref:`here <executor-types-comparison>` for comparisons of executors). Running multiple executors allows you to make better use of the strengths of all the available executors and avoid their weaknesses. In other words, you can use a specific executor for a specific set of tasks where its particular merits and benefits make the most sense for that use case.
+
+Configuration
+^^^^^^^^^^^^^
+
+Configuring multiple executors uses the same configuration option (as described :ref:`here <executor-basic-configuration>`) as single executor use cases, leveraging a comma separated list notation to specify multiple executors.
+
+.. note::
+    The first executor in the list (either on its own or along with other executors) will behave the same as it did in pre-2.10.0 releases. In other words, this will be the default executor for the environment. Any Airflow Task or DAG that does not specify a specific executor will use this environment level executor. All other executors in the list will be initialized and ready to run tasks if specified on an Airflow Task or DAG. If you do not specify an executor in this configuration list, it cannot be used to run tasks.
```

Comment:
Should we say something about the preciously mixed executors? `CeleryKubernetesExecutor` and others?

I would imagine that some people use them and they might consider migrating to this feature. Do we need to warn about cases of

`executor = 'CeleryKubernetesExecutor,EcsExecutor'`

Reply:
There is a section dedicated to the previously mixed/hybrid executors. Does that suffice?

> Do we need to warn about cases of

> executor = 'CeleryKubernetesExecutor,EcsExecutor'

Other than it being an odd choice, I don't think there's anything that will break if users try to do this. The section I wrote about the old hybrid executors makes it clear that we discourage their use, but until they're deprecated, we can't really stop folks from using them.